### PR TITLE
MCSCF module small improvements

### PR DESCRIPTION
**Pull request by bogdanoff**
- ~~Check the largest orbital rotation and if it is small ~pass fciRestart=true to fcisolver~ set ci0=True for external solvers~~
- Add minimal value for stepsize
- Add option to use state-average 1rdm for canonicalization
- Pass wfnsym in addon functions
- print symmetry information of the active space at _info_ level, such that user know what is going on

### CODE REVIEW

File: pyscf/mcscf/casci.py [hunk 1]
```diff
@@ -412,10 +414,20 @@ def canonicalize(mc, mo_coeff=None, ci=None, eris=None, sort=False,
...
-                     'used to compute the natural orbitals in active space.')
-            casdm1 = mc.fcisolver.make_rdm1(ci[0], mc.ncas, mc.nelecas)
+                not isinstance(mc.fcisolver, addons.StateAverageFCISolver)):
+            if stav_dm1:
+                log.warn('Mulitple states found in CASCI solver. '
+                         'Use state-average 1RDM  to compute the Fock matrix'
+                         ' and natural orbitals in the active space.')
+                casdm1 = mc.fcisolver.make_rdm1(ci[0], mc.ncas, mc.nelecas)
+                for root in range(1, len(ci)):
+                    casdm1 += mc.fcisolver.make_rdm1(ci[root], mc.ncas,
```
[the remark this answers is not in the document]
Comment (bogdanoff) [reply pyscf/mcscf/casci.py:424]: Yes, this actually activated if one asks FCI solver for more then one root in CASCI calculation. Before the only option was to use the first root during the canonicalization. My idea was to activate the state-average densities in this case, or at least have an option to do it.

File: pyscf/mcscf/casci.py [hunk 2]
```diff
@@ -514,7 +526,7 @@ def kernel(casci, mo_coeff=None, ci0=None, verbose=logger.NOTE):
     e_tot, fcivec = casci.fcisolver.kernel(h1eff, eri_cas, ncas, nelecas,
                                            ci0=ci0, verbose=log,
                                            max_memory=max_memory,
-                                           ecore=energy_core)
+                                           ecore=energy_core, fciRestart=fciRestart)
```
[the remark this answers is not in the document]
Comment (bogdanoff) [reply pyscf/mcscf/casci.py]: Thanks for the suggestion, I think the new code is better in this respect. With ci0 a special treatment of state-average solvers is required, so I had to import the state-average class from mcscf.addons to check for that.

**Review by sunqm — APPROVED** [review]

File: pyscf/mcscf/casci.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdanoff) [reply pyscf/mcscf/casci.py:424]: Interestingly, I saw the message about using the ground state 1-pdm in the output of some tests.

**Comment by bogdanoff** [comment]
All tests are passing now. The biggest problem was that somewhat larger starting stepsize value (0.05) causes too strong deviations, which probably should not be the case.

**Comment by bogdanoff** [comment]
I removed the restart due to small orbital rotation functionality, such that the rest of the changes can be merged in.

**Review by sunqm — APPROVED** [review]

**Comment by sunqm** [comment]
Moved to PR #1182

**Pull request closed without merging**